src/nft/components/card/containers.tsx

```diff
@@ -0,0 +1,849 @@
+`
+
+const SecondaryRow = ({ children }: { children: ReactNode }) => <StyledSecondaryRow>{children}</StyledSecondaryRow>
+
+const StyledSecondaryDetails = styled(Row)`
+  overflow: hidden;
+  white-space: nowrap;
+`
+
+const SecondaryDetails = ({ children }: { children: ReactNode }) => (
+  <StyledSecondaryDetails>{children}</StyledSecondaryDetails>
+)
+
+const SecondaryInfoContainer = styled(ThemedText.BodyPrimary)`
+  color: ${({ theme }) => theme.textPrimary};
+  overflow: hidden;
+  white-space: nowrap;
+  text-overflow: ellipsis;
+  line-height: 24px;
+`
+
+const SecondaryInfo = ({ children }: { children: ReactNode }) => {
+  return <SecondaryInfoContainer>{children}</SecondaryInfoContainer>
+}
+
+interface RankingProps {
+  provider: { url?: string; rank?: number }
+}
+
+const RarityLogoContainer = styled(Row)`
+  margin-right: 8px;
+  width: 16px;
+`
+
+const RarityText = styled(ThemedText.BodySmall)`
+  display: flex;
+`
+
+const RarityInfo = styled(ThemedText.Caption)`
+  flex-shrink: 0;
+  color: ${({ theme }) => theme.textSecondary};
+  background: ${({ theme }) => theme.backgroundInteractive};
+  padding: 4px 6px;
+  border-radius: 4px;
+  font-weight: 700 !important;
+  line-height: 12px;
+  text-align: right;
+  cursor: pointer;
+`
+
+const Ranking = ({ provider }: RankingProps) => {
+  if (!provider.rank) {
+    return null
+  }
+
+  return (
+    <RarityInfo>
+      <MouseoverTooltip
+        text={
+          <Row>
+            <RarityLogoContainer>
+              <img src="/nft/svgs/gem.svg" width={16} height={16} />
+            </RarityLogoContainer>
+            <RarityText>Ranking by Rarity Sniper</RarityText>
+          </Row>
+        }
+        placement="top"
+      >
+        # {putCommas(provider.rank)}
+      </MouseoverTooltip>
+    </RarityInfo>
+  )
+}
+
+const SUSPICIOUS_TEXT = t`Blocked on OpenSea`
+
+const SuspiciousIconContainer = styled(Row)`
+  flex-shrink: 0;
+`
+
+const Suspicious = () => {
+  return (
+    <MouseoverTooltip text={<ThemedText.BodySmall>{SUSPICIOUS_TEXT}</ThemedText.BodySmall>} placement="top">
+      <SuspiciousIconContainer>
+        <SuspiciousIcon />
+      </SuspiciousIconContainer>
+    </MouseoverTooltip>
+  )
+}
+
+const StyledMarketplaceContainer = styled.div<{ isText?: boolean }>`
+  position: absolute;
+  display: flex;
+  top: 12px;
+  left: 12px;
+  height: 32px;
+  width: ${({ isText }) => (isText ? 'auto' : '32px')};
+  padding: ${({ isText }) => (isText ? '0px 8px' : '0px')};
+  background: rgba(93, 103, 133, 0.24);
+  justify-content: center;
+  align-items: center;
+  border-radius: 32px;
+  z-index: 2;
+`
+
+const ListPriceRowContainer = styled(Row)`
+  gap: 6px;
+  color: ${({ theme }) => theme.accentTextLightPrimary};
+  font-size: 14px;
+  font-weight: 600;
+  line-height: 16px;
+`
+
+function getMarketplaceIcon(market: Markets): ReactNode {
+  switch (market) {
+    case Markets.Opensea:
+      return <OpenSeaMarketplaceIcon width="20px" height="20px" />
```

Comment:
could set `width="20px" height="20px" ` as a const args and use that for all icons